src/sentry/search/snuba/backend.py

```diff
@@ -219,43 +231,73 @@ def _query(self, project, retention_window_start, group_queryset, tags, environm
                 parameters,
             )
 
-        # TODO: If the query didn't include anything to significantly filter
-        # down the number of groups at this point ('first_release', 'query',
-        # 'status', 'bookmarked_by', 'assigned_to', 'unassigned',
-        # 'subscribed_by', 'active_at_from', or 'active_at_to') then this
-        # queryset might return a *huge* number of groups. In this case, we
-        # probably *don't* want to pass candidates down to Snuba, and rather we
-        # want Snuba to do all the filtering/sorting it can and *then* apply
-        # this queryset to the results from Snuba.
-        #
-        # However, if this did filter down the number of groups significantly,
-        # then passing in candidates is, of course, valuable.
-        #
-        # Should we decide which way to handle it based on the number of
-        # group_ids, the number of hashes? Or should we just always start the
-        # query with Snuba? Something else?
-        candidate_group_ids = list(group_queryset.values_list('id', flat=True))
-
-        sort, extra_aggregations, calculate_cursor_for_group = sort_strategies[sort_by]
-
-        group_data = do_search(
+        candidate_hashes = None
+        if MAX_PRE_SNUBA_CANDIDATES:
+            # If the query didn't include anything to significantly filter
+            # down the number of group hashes at this point ('first_release', 'query',
```

Comment:
Isn't the "the number of group hashes" here "the number of groups" (though I guess more groups also implies more hashes.)

Reply:
I don't think so? It's a `GroupHash` query now (with a `Group` subquery/join inside).

Note that this may change by the time you read this because I need to figure out how to handle `GroupHashTombstone`s here, so I may revert to using `Group` (and passing it down as an `issue` to the Snuba utils).